pkg/k8sMetadataUtil/k8sMetadataUtil.go

```diff
@@ -49,17 +54,23 @@ func getClientSet() (*kubernetes.Clientset, error) {
 	return clientset, nil
 }
 
-func setNodesCount(clientset *kubernetes.Clientset) {
-
-	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+func getClusterUuid(clientset *kubernetes.Clientset) (types.UID, error) {
+	clusterMetadata, err := clientset.CoreV1().Namespaces().Get(context.TODO(), "kube-system", metav1.GetOptions{})
 	if err != nil {
-		errString := fmt.Sprintf("List error: %s", err.Error())
-		fmt.Println(errString)
-		nodesCount = -1
-		nodesCountErr = errors.New(errString)
-		return
+		return "", err
```

Comment:
What is the chance that the cluster uuid will change every hour? maybe we can query it just once